samples/excel/85-preview-apis/data-types-formatted-number.yaml

```diff
@@ -0,0 +1,149 @@
+order: 1
+id: excel-data-types-formatted-number
+name: 'Data types: Formatted numbers'
+description: This sample shows how to set and get data types using the formatted number properties.
```

Comment:
Meta question for the feature: why would I opt to use the formatted number data type instead of setting `values` and `numberFormats` separately?

Reply:
The `numberFormat` property only controls the display of a value in the Excel UI. It doesn't change the underlying number value, even when the value changes dramatically through formatting as a date or currency. The discrepancy between the display and the underlying value can lead to errors during calculation and confusion for the end user. I think the short answer is that formatting with data types is safer, and once released to GA will be the recommended strategy for number formatting with add-ins. This is described in the conceptual docs. Do you think it should be addressed directly in the Script Lab sample, too?

Reply:
Maybe I'm misunderstanding the question. My answer assumed you were referring to [Excel.NumberFormatInfo class](https://docs.microsoft.com/en-us/javascript/api/excel/excel.numberformatinfo?view=excel-js-preview), which is a class and not a property (my mistake).

Reply:
I was referring to [Range.numberFormat](https://docs.microsoft.com/en-us/javascript/api/excel/excel.range?view=excel-js-preview#excel-excel-range-numberformat-member). That's another way to set the format without changing the value.